### ISSUE TITLE
Qustion with regard to the different learning ability between models

### ISSUE THREAD

**Issue by PosoSAgapo** [DannyMerkx/next_word_prediction#1]
I notice that in your work, you have made it possible that parameters are comparable for 2 different models. 
However, from what I understand, the same amount of parameter does not equal the same ability of learning. 
If that's the case, a simple MLP layer could be stacked to compete with SOTA models in CV or NLP, but as far as we know, this is not true since we do know that some models learns well because of a more complex and efficient structure.
The transformer model itself is just much more powerful comparing to exsiting recurrent models like GRU, BiLSTM even with same amount of parameters. If a new recurrent model in the future could perform like transformer, does the idea of 'human reading style is like attention rather than recurrent'  still hold?

**Comment by DannyMerkx** [comment]
Hi PosoSAgapo,

It's true that a comparable number of parameters does guarantee equal learning capability. For a fair comparison between models that is however one of the things we did for a starting point, but the most important thing is making sure that the predictive power is equal as well (as measures by the loss and average surprisal over the test stimuli). From that starting point we let go of keeping the parameters equal and found that increasing depth did not improve modelling capability of the RNNs but did for the Transformer. 

I think the final question is a hard one to answer. The 'best' model in explaining human experimental data is only ever tangential evidence that human processing maybe involves something akin to what the model does but I think this kind of evidence alone can ever conclusively prove it to be so. So, maybe indeed at some point a change in recurrent architecture outperforms the transformer and were back where we started in this regard. In the meantime, RNNs successes as a computational model have inspired lots of modelling research providing evidence from other angles and at the moment Transformers are undergoing the same process, reviving interest for instance in 'Cue based retrieval' models where attention like processing might actually fit the theory.  So at the least something new is getting attention now and when people only ever look at RNNs we'll only find evidence related to RNNs. So this 'human reading style is like attention rather than recurrent' debate needs well developed theories on both sides of the argument.

Thanks for your interest, let me know if this answers your questions about our work.

**Comment by PosoSAgapo** [comment]
> Hi PosoSAgapo,
> 
> It's true that a comparable number of parameters does guarantee equal learning capability. For a fair comparison between models that is however one of the things we did for a starting point, but the most important thing is making sure that the predictive power is equal as well (as measures by the loss and average surprisal over the test stimuli). From that starting point we let go of keeping the parameters equal and found that increasing depth did not improve modelling capability of the RNNs but did for the Transformer.
> 
> I think the final question is a hard one to answer. The 'best' model in explaining human experimental data is only ever tangential evidence that human processing maybe involves something akin to what the model does but I think this kind of evidence alone can ever conclusively prove it to be so. So, maybe indeed at some point a change in recurrent architecture outperforms the transformer and were back where we started in this regard. In the meantime, RNNs successes as a computational model have inspired lots of modelling research providing evidence from other angles and at the moment Transformers are undergoing the same process, reviving interest for instance in 'Cue based retrieval' models where attention like processing might actually fit the theory. So at the least something new is getting attention now and when people only ever look at RNNs we'll only find evidence related to RNNs. So this 'human reading style is like attention rather than recurrent' debate needs well developed theories on both sides of the argument.
> 
> Thanks for your interest, let me know if this answers your questions about our work.

Yes, I do acknowledege this is the best we can do for know. I am just curious about this, since some of my research are also related with Human Sentence Processing model. It is ture that 2 models like Suprisal  and Working memory all assume that human process sentences in a incremental/recurrent way, it is just surprising to see that Transformers could break this widely accepted rules.
